CMakeLists.txt

```diff
@@ -108,3 +108,34 @@ install(
     FILES "${CMAKE_CURRENT_BINARY_DIR}/wpebackend-fdo-${WPEBACKEND_FDO_API_VERSION}.pc"
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
 )
+
+option(BUILD_DOCS "Build the documentation" OFF)
+if (BUILD_DOCS)
+    find_program(HOTDOC hotdoc)
+    if (NOT HOTDOC)
+        message(FATAL_ERROR "hotdoc not found!")
+    endif ()
+
+    execute_process(
+        COMMAND ${HOTDOC} --has-extension c-extension
+        RESULT_VARIABLE HAS_HOTDOC_C_EXTENSION
+    )
+    if ("${HAS_HOTDOC_C_EXTENSION}" EQUAL 0)
+        add_custom_target(Documentation ALL
+            ${HOTDOC} run
+                --project-name WPEBackend-fdo
+                --project-version ${PROJECT_VERSION}
+                --sitemap ${CMAKE_SOURCE_DIR}/docs/sitemap.txt
+                --output ${CMAKE_CURRENT_BINARY_DIR}/Documentation/
+                --c-smart-index
+                --extra-c-flags=-DWPE_FDO_COMPILATION
+                --c-sources
+                    ${CMAKE_SOURCE_DIR}/include/wpe-fdo/*.h
```

Comment:
This is where the GtkDoc comment will be looked for, do you plan on documenting in the headers? (I prefer doing so personnaly, but I just want to make sure it is your intents here).

Reply:
Personally I *do* like documentation in headers, because if I am missing something and I don't have the documentation at hand there's also the possibility of opening the installed `.h` files to take a look.

I am aware that most projects which follow GLib/GNOME conventions put the documentation comments in the `.c` files, so I am fine with that as well if other prefer it.

Reply:
+2 for doc in headers then :-)